cluster.go

```diff
@@ -82,15 +84,26 @@ type clusterClient struct {
 	aws    bool
 }
 
+// NOTE: clusterConnection and conn must be initialized at the same time
+type clusterConnection struct {
+	conn              conn
+	isPrimaryNodeConn bool
+}
+
 func newClusterClient(opt *ClientOption, connFn connFn) (client *clusterClient, err error) {
 	client = &clusterClient{
 		cmd:    cmds.NewBuilder(cmds.InitSlot),
 		opt:    opt,
 		connFn: connFn,
-		conns:  make(map[string]conn),
+		conns:  make(map[string]*clusterConnection),
```

Comment:
```suggestion
		conns:  make(map[string]connrole),
```
clusterConnection is too long for me. And I think it is better to use struct directly.

Reply:
[3bb4fb5](https://github.com/redis/rueidis/pull/402/commits/3bb4fb58e43879fa87824249b8325a4751323c7d) OK. I changed it.